lib/poseidon/messages_for_broker.rb

```diff
@@ -36,14 +36,20 @@ def build_protocol_objects(compression_config)
       end
     end
 
-    # We can always retry these errors because they mean none of the kafka brokers persisted the message
-    ALWAYS_RETRYABLE = [Poseidon::Errors::LeaderNotAvailable, Poseidon::Errors::NotLeaderForPartition]
+    # Put any errors here that can result in messages persisted by one or more Kafka brokers
+    NOT_RETRYABLE = [
+      # Currently none
+    ]
 
     def successfully_sent(producer_response)
       failed = []
       producer_response.topic_response.each do |topic_response|
         topic_response.partitions.each do |partition|
-          if ALWAYS_RETRYABLE.include?(partition.error_class)
+          if NOT_RETRYABLE.include?(partition.error_class)
+            Poseidon.logger.debug { "Received #{partition.error_class} when attempting to send messages to #{topic_response.topic} on #{partition.partition}" }
+            # TODO: For not retryable errors, we should disable the retrying,
+            # yet still return an error.
```

Comment:
The TODO isn't urgent and can be handled at the point when we discover an error that shouldn't be retried (I'm not aware of any)